### ISSUE TITLE
module s3.get does not support eu-central-1

### ISSUE THREAD

**Issue by ekle** [saltstack/salt#20518]
the downloaded file does only contain an error message:

    <Error><Code>InvalidRequest</Code><Message>The authorization mechanism you have provided is not supported. Please use AWS4-HMAC-SHA256.</Message><RequestId>....</RequestId><HostId>...</HostId></Error>

anonymous access does also not work

**Comment by jfindlay** [comment]
@ekle, will you provide the output of `salt --versions-report`?  Thanks.

**Comment by ekle** [comment]
here:

    salt  --versions-report
        Salt: 2014.7.1
        Python: 2.7.8 (default, Oct 18 2014, 12:50:18)
        Jinja2: 2.7.3
        M2Crypto: 0.21.1
        msgpack-python: 0.4.2
        msgpack-pure: Not Installed
        pycrypto: 2.6.1
        libnacl: Not Installed
        PyYAML: 3.11
        ioflo: Not Installed
        PyZMQ: 14.4.0
        RAET: Not Installed
        ZMQ: 4.0.5
        Mako: 1.0.0

**Comment by yannick** [comment]
same here for salt-cloud, which lib is used for auth, i guess its the V4 signature thing.


```[root@sm1 salt]# salt-cloud --versions-report
            Salt: 2014.7.4
          Python: 2.7.9 (default, Dec 11 2014, 04:42:00)
          Jinja2: 2.7.3
        M2Crypto: 0.22
  msgpack-python: 0.4.6
    msgpack-pure: Not Installed
        pycrypto: 2.6.1
         libnacl: Not Installed
          PyYAML: 3.11
           ioflo: Not Installed
           PyZMQ: 14.5.0
            RAET: Not Installed
             ZMQ: 4.0.5
            Mako: Not Installed
 Apache Libcloud: 0.16.0
```

**Comment by slinu3d** [comment]
Created a pull request https://github.com/saltstack/salt/pull/23467 adding AWS v4 and it should fix this issue.

**Comment by rallytime** [comment]
S3 was moved to AWS sig4 also with #25420. I think that should resolve this issue, and the fix will be available in 2015.5.4. Is anyone here in a position to test with the HEAD of 2015.5 to confirm this is resolved?

**Comment by rallytime** [comment]
Since we didn't receive a response here and 2015.5.5 has been released, I am going to go ahead and close this. If this pops up again, certainly leave a comment and we can reopen and take another look. Thanks!
